src/mito2/src/worker/handle_alter.rs

```diff
@@ -280,3 +282,16 @@ fn log_option_update<T: std::fmt::Debug>(
         option_name, region_id, prev_value, cur_value
     );
 }
+
+/// Used to determine whether we can build index directly after schema change.
+fn can_fast_path_index(kind: &AlterKind) -> bool {
+    match kind {
+        // `SetIndexes` is a fast-path operation because it can build indexes for existing SSTs
+        // in the background, without needing to wait for a flush or compaction cycle.
+        AlterKind::SetIndexes { options: _ } => true,
+        // For AddColumns, DropColumns, UnsetIndexes and ModifyColumnTypes, we don't treat them as index changes.
+        // Index files still need to be rebuilt after schema changes,
+        // but this will happen automatically during flush or compaction.
+        _ => false,
+    }
+}
```

Comment:
The meaning of `fast path index` is a little confusing. Maybe `need change index` is more straightforward.